src/sentry/static/sentry/app/components/forms/selectField.jsx

```diff
@@ -42,7 +43,7 @@ export default class SelectField extends FormField {
       //
       // This can happen when this is apart of a field, and it re-renders onChange for a different field,
       // there will be a mismatch between this component's state.value and `this.getValue` result above
-      if (newValue !== coercedValue && !!newValue !== !!coercedValue) {
+      if (!isEqual(newValue, coercedValue) && !!newValue !== !!coercedValue) {
```

Comment:
I don't think this is the problem, when testing it locally it still doesn't reset.

I think the issue is with the 2nd condition, it's there so that we don't infinitely loop when newValue is null and coercedValue is ''. Let's remove that condition and add an early return when `!newValue && !coercedValue`.

"clearable" seems to cause some other weird issues too (e.g. select something in Order By, click the "x" to clear, it goes to the default value `timestamp`, then if you clear again, it becomes empty).

Reply:
@billyvg Have a look at the failing test, it seems like setting the value of one field somehow impacts the other 😱